web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte

```diff
@@ -32,33 +32,26 @@
 
   let { isViewing: showAssetViewer } = assetViewingStore;
   const assetStore = new AssetStore({ isArchived: false, withStacked: true, withPartners: true });
-  const assetInteractionStore = createAssetInteractionStore();
-  const { isMultiSelectState, selectedAssets } = assetInteractionStore;
+  const assetInteraction = new AssetInteraction();
 
-  let isAllFavorite = $state(false);
-  let isAllOwned = $state(false);
-  let isAssetStackSelected = $state(false);
-  let isLinkActionAvailable = $state(false);
-
-  $effect(() => {
-    const selection = [...$selectedAssets];
-    isAllOwned = selection.every((asset) => asset.ownerId === $user.id);
-    isAllFavorite = selection.every((asset) => asset.isFavorite);
-    isAssetStackSelected = selection.length === 1 && !!selection[0].stack;
-    const isLivePhoto = selection.length === 1 && !!selection[0].livePhotoVideoId;
+  let selectedAssets = $derived(assetInteraction.selectedAssetsArray);
+  let isAllOwned = $derived(selectedAssets.every((asset) => asset.ownerId === $user.id));
+  let isAssetStackSelected = $derived(selectedAssets.length === 1 && !!selectedAssets[0].stack);
+  let isLinkActionAvailable = $derived.by(() => {
+    const isLivePhoto = selectedAssets.length === 1 && !!selectedAssets[0].livePhotoVideoId;
     const isLivePhotoCandidate =
-      selection.length === 2 &&
-      selection.some((asset) => asset.type === AssetTypeEnum.Image) &&
-      selection.some((asset) => asset.type === AssetTypeEnum.Image);
-    isLinkActionAvailable = isAllOwned && (isLivePhoto || isLivePhotoCandidate);
-  });
+      selectedAssets.length === 2 &&
+      selectedAssets.some((asset) => asset.type === AssetTypeEnum.Image) &&
+      selectedAssets.some((asset) => asset.type === AssetTypeEnum.Image);
```

Comment:
I realize that this has been duplicated before as well. But why? Am I overlooking something or are those the exact same conditions?

Reply:
This is indeed wrong, I'll open a PR to fix it